internal/app/machined/pkg/system/services/extension.go

```diff
@@ -64,6 +67,13 @@ func (svc *Extension) PostFunc(r runtime.Runtime, state events.ServiceState) (er
 func (svc *Extension) Condition(r runtime.Runtime) conditions.Condition {
 	conds := []conditions.Condition{}
 
+	if svc.Spec.Container.EnvironmentFile != "" {
+		// add a dependency on the environment file
+		svc.Spec.Depends = append(svc.Spec.Depends, extservices.Dependency{
```

Comment:
this mutates the extension spec, and potentially multiple times, I think it's much better to build directly a `conditions.WaitForFileToExist` here

Reply:
fixed